Lib/test/datetimetester.py

```diff
@@ -1350,6 +1350,17 @@ def test_strftime(self):
         #check that this standard extension works
         t.strftime("%f")
 
+        # bpo-35066: make sure trailing '%' don't cause
+        # datetime's strftime to complain
+        failed = False
+        try:
+            t.strftime('%')
+        except ValueError:
+            failed = True
+        if not failed:
```

Comment:
I would prefer this to be in a subtest, like so:

```python
with self.subTest('trailing percentage'):
    if not time_supports_trailing_percent:
        self.skipTest('time module does not support trailing %')

    self.assertEqual(t.strftime('%'), '%'))
    self.assertEqual(t.strftime("m:%m d:%d y:%y %"), "m:03 d:02 y:05 %")
```

Though I'm thinking maybe this should instead be moved to its own test method. Partially because I tend to prefer a larger number of small, targeted test methods rather than big mega-tests and partially because `skipTest` and `subTest` when used together apparently give a misleading message. An example:

```python
import unittest

class SomeTest(unittest.TestCase):
    def test_something(self):
        for i in range(1, 3):
            with self.subTest(i):
                if i > 1:
                    self.skipTest('Not supported')

                self.assertEqual(i, 1)
```

When you run this with `python -m unittest -v`, you get:

```
$ python -m unittest -v
test_something (test_mod.SomeTest) ... skipped 'Not supported'

----------------------------------------------------------------------
Ran 1 test in 0.000s

OK (skipped=1)
```

And with `pytest` you get:

```
========= test session starts =======

platform linux -- Python 3.7.1, pytest-4.0.1, py-1.7.0, pluggy-0.8.0
rootdir: /tmp/test_mod, inifile:
collected 1 item

test_mod.py s                                           [100%]

===== 1 skipped in 0.00 seconds =====
```

Both are misleading, because only the `i == 2` subtest is being skipped *not* the whole test.

Reply:
Ok, so parking the test in it's own method will allow me to skip it on a failing `time` call without making it look like all the other tests in that method were skipped.